### PR TITLE
Make ANSI colors data in ansi_up constant

### PR DESCRIPTION
**Pull request by osa1**
The `ansiColors` array in `AnsiUp` is constant data and it doesn't need to be
instantiated with every instance of `AnsiUp`.

Move it to the top-level as `const`.

Also initialize `palette256` once, instead of on every instantiation of `AnsiUp`.

### CODE REVIEW

**Review by kenzieschmoll — COMMENTED** [review]
please bump the ansi_up package version and add a changelog entry

**Comment by osa1** [comment]
@kenzieschmoll done, PTAL.

Currently it looks like we need to publish this on pub.dev before being able to use it in devtools, right?

https://github.com/flutter/devtools/blob/792ba0e10a0e1d4d4154653cd5515b0d0b8cffe8/packages/devtools_app/pubspec.yaml#L16

I wonder if we should start using the version in the tree instead, that way we don't have to publish every change before using it in devtools.

(Also, the changelog CI check is still failing because it wants a changelog entry in the devtools_app changelog..)

**Comment by osa1** [comment]
> I wonder if we should start using the version in the tree instead, that way we don't have to publish every change before using it in devtools.

I've addressed this in #8251.

File: third_party/packages/ansi_up/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
+## 2.0.0-dev
```
Comment (kenzieschmoll) [inline third_party/packages/ansi_up/CHANGELOG.md:1]: can you please merge in the changes from https://github.com/flutter/devtools/commit/168f4ec292d17a6f56942e5a880e25e039c54893?

**Pull request merged**